util/util.go

```diff
@@ -493,15 +494,30 @@ func ClusterToObjectsMapper(c client.Client, ro runtime.Object, scheme *runtime.
 		return nil, err
 	}
 
+	isNamespaced, err := isAPINamespaced(gvk, c.RESTMapper())
```

Comment:
So in prod we can check it this way. Unfortunately, the fake client doesn't have a rest mapper implementation. 

Not sure what the best way forward is. 

Do we want to have this detection? If yes, is it reasonable to implement a restmapper in the fakeclient?

Reply:
It's probably fine to keep the check and potentially implement a fake rest mapper